src/pages/DistributePage/DistributePage.test.tsx

```diff
@@ -0,0 +1,94 @@
+import { act, render, screen, waitFor } from '@testing-library/react';
+
+import { TestWrapper } from 'utils/testing';
+import { mockVault, mockEpoch } from 'utils/testing/mocks';
+
+import DistributePage from './DistributePage';
+
+jest.mock('hooks/gql/useCurrentOrg', () => ({
+  useCurrentOrg: jest.fn(() => [1, jest.fn()]),
+}));
+
+jest.mock('react-router-dom', () => {
+  const library = jest.requireActual('react-router-dom');
+  return {
+    useParams: {
+      epochId: '5',
+    },
+    ...library,
+  };
+});
+
+jest.mock('hooks/gql/useVaults', () => ({
+  useVaults: () => ({
+    data: [mockVault],
+  }),
+}));
+
+jest.mock('./queries', () => ({
+  usePreviousDistributions: jest.fn().mockImplementation(() => ({
+    data: {
+      id: 2,
+      vault_id: 2,
+      distribution_json: {},
+    },
+  })),
+  useCurrentUserForEpoch: async () => ({
+    users: {
+      id: 21,
+      name: 'Mock User',
+      address: '0x0',
+      role: 1,
+      circle_id: 2,
+    },
+  }),
+  useSubmitDistribution: jest.fn(),
+  useGetAllocations: jest
+    .fn()
+    .mockReturnValue({
+      data: mockEpoch,
+    })
+    .mockReturnValueOnce({
+      data: mockEpoch,
+    })
+    .mockReturnValueOnce({
+      data: null,
+    }),
+}));
```

Comment:
Would be nice if this was moved to a dedicated fixtures files so it could be shared between multiple page tests.

Reply:
these are pretty specific to this page; i think it's unlikely to be very reusable. maybe just the mock user & epoch? i have some similar mocks in `utils/testing/recoil` that might be reusable for this purpose